router/router.go

```diff
@@ -1742,9 +1753,8 @@ func (rt *HandleT) generatorLoop(ctx context.Context) {
 
 			generatorStat.Start()
 
-			processCount := rt.readAndProcess()
-
-			countStat.Count(processCount)
+			rt.cleanUp()
```

Comment:
It is unsafe to perform this operation here, since we are introducing a(nother) new async stage through the `readProcessChan`. Please move it to its original place, as the first step of the `processRouterJobs` method which is the one that decides (synchronously) what enters the worker channel and what not.
If possible, don't extract the if block in its own method, simply add it in the method.

FYI, I am refactoring the whole event ordering algorithm in #2344